### DISCUSSION TITLE
SNMPv3 - Protokollauswahl in WebUI

### DISCUSSION THREAD

**Discussion by KoMa1978** [OpenCCU/OpenCCU#2687] [🇩🇪 Ideen]
Hallo zusammen,

in #2673 wurden stärkere SNMP Protokolle implementiert.
Das funktioniert generell auch, allerdings nur per CLI in den Konfig-Dateien, was in meinen Tests recht fehleranfällig war.

Ich fände es daher gut und sinnvoll, die Konfiguration von PRIV und AUTH Protokoll in der WEB-Oberfläche als DropDown zu implementieren. Alle notwendigen Infos (OIDs und Bezeichner für createUser Option) sind im #2673 dokumentiert.

Besten Dank,
 KoMa1978

**Comment by jens-maus** [comment]
Also meiner Meinung nach ist SNMP im Grunde total overengineered, viel zu kompliziert in der Nutzung und eigentlich auch recht obsolete, alsdass sich hier noch irgendwie substantiell lohnt hier irgendwie Arbeit reinzustecken. Vor allem hier irgendwelche WebUI Anpassungen für das setzen von irgendwelchen Settings des SNMP umzusetzen ist im Grunde vergebene Liebesmühe. 

Gerne bin ich aber natürlich bereit irgendwelche PullRequests hierzu entgegen zu nehmen falls es hier jemand als doch wichtig erachtet die SNMP Unterstützung in RaspberryMatic noch weiter anzupassen.

**Comment by Baxxy13** [comment]
Was super wäre, wäre eine MIB für RaspberryMatic zu haben.
Ich hatte ja mal ein bisschen mit SNMP experimentiert um Statusdaten der RM in Homeassistant zu bekommen.
Funktioniert grundlegend sehr gut, aber ist mangels Dokumentation der Parameter ziemlich aufwendig.

**Reply by jens-maus** [reply]
Da es hier nix RaspberryMatic spezielles gibt müsste/sollte ein default Linux MIB eigentlich reichen um den Status des RaspberryMatic Systems abzuprüfen, etc. Alles darüberhinaus (z.B. ganze HomeMatic Geräte via SNMP durchzuleiten) ist IMHO nicht gerade sinnvoll.

**Reply by KoMa1978** [reply]
Da schließe ich mich an ... aktuell geht der Trend ja scheinbar in die Richtung auf den Geräten "Exporter" zu installieren, die die Metriken dann liefern. Und man könnte es ja heute auch schon wunderbar per XMLRPC abgreifen, wie es andere Tools (z.B. iobroker) auch machen.

Zur Not könnte man SNMP auch mittels EXTEND erweitern, macht aber m.E. wenig Sinn.

Mir geht es hier nur um aktuelle/starke Algorithmen für SNMP. Leider bin ich kein Programmierer und kann daher (bis auf diese grundlegeneden Infos in #2673 nicht viel beitragen.

**Reply by Baxxy13** [reply]
Mir geht's primär um grundlegende Statusdaten vom OS, wie Load, free Mem usw.
Da muss ich mich bei Gelegenheit mal wieder mit beschäftigen ob eine Standard-MIB da Aufschluss bringt.
Aktuell parse ich die Monit Ausgabe im XML-Format, funktioniert auch.

**Reply by KoMa1978** [reply]
Das ist tatsächlich Standard Linux und wenn du da PRTG, Icinga oder was auch immer nutzt und ein vorgefertigtes Template nimmst, sollte das wunderbar hin hauen.

Die MIBs machen ja auch nichts anderes, als die OID in "lesbareren/verständlicheren" Text umzuwandeln.
